modules/tensor_mechanics/src/actions/GeneralizedPlaneStrainAction.C

```diff
@@ -116,7 +130,24 @@ GeneralizedPlaneStrainAction::act()
     std::string uo_type = "GeneralizedPlaneStrainUserObject";
     InputParameters params = _factory.getValidParams(uo_type);
 
-    params.applyParameters(parameters());
+    // Skipping selected parameters in applyParameters() and then manually setting them only if they
+    // are set by the user is just to prevent both the current and deprecated variants of these
+    // parameters from both getting passed to the UserObject. Once we get rid of the deprecated
+    // versions, we can just set them all with applyParameters().
+    params.applyParameters(
+        parameters(),
+        {"out_of_plane_pressure", "out_of_plane_pressure_function", "factor", "pressure_factor"});
+    if (parameters().isParamSetByUser("out_of_plane_pressure"))
+      params.set<FunctionName>("out_of_plane_pressure") =
+          getParam<FunctionName>("out_of_plane_pressure");
+    if (parameters().isParamSetByUser("out_of_plane_pressure_function"))
+      params.set<FunctionName>("out_of_plane_pressure_function") =
+          getParam<FunctionName>("out_of_plane_pressure_function");
```

Comment:
You mentioned that having a default value automatically makes `isParamSetByUser` return true. So wouldn't this condition always be fulfilled?

Reply:
Oh yeah, good catch!  I need to remove the default value for `out_of_plane_pressure_function` in this class.